Auto update failing for ninja

Hi, I'm Termux 🤖.

I'm here to help you update your Termux packages.

I've tried to update the [ninja](https://github.com/termux/termux-packages/blob/9228af3dbf0eb99eb84a389aa8ff41b5ca8d7945/packages/ninja) package, but it failed.

Here's the output of the update script:
<details>
<summary>Show log</summary>
<pre lang="bash">
INFO: Updating ninja [Current version: 1.11.1]
INFO: package being updated to 1.12.0.

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0

100 13167    0 13167    0     0  44004      0 --:--:-- --:--:-- --:--:-- 44004
100  234k    0  234k    0     0   623k      0 --:--:-- --:--:-- --:--:-- 2918k

You are about to commit these changes:

--------------------
diff --git a/packages/ninja/build.sh b/packages/ninja/build.sh
index e5dbab595..259b9d59b 100644
--- a/packages/ninja/build.sh
+++ b/packages/ninja/build.sh
@@ -2,9 +2,9 @@ TERMUX_PKG_HOMEPAGE=https://ninja-build.org
 TERMUX_PKG_DESCRIPTION="A small build system with a focus on speed"
 TERMUX_PKG_LICENSE="Apache-2.0"
 TERMUX_PKG_MAINTAINER="@termux"
-TERMUX_PKG_VERSION="1.11.1"
+TERMUX_PKG_VERSION="1.12.0"
 TERMUX_PKG_SRCURL=https://github.com/ninja-build/ninja/archive/v${TERMUX_PKG_VERSION}.tar.gz
-TERMUX_PKG_SHA256=31747ae633213f1eda3842686f83c2aa1412e0f5691d1c14dbbcc67fe7400cea
+TERMUX_PKG_SHA256=8b2c86cd483dc7fcb7975c5ec7329135d210099a89bc7db0590a07b0bbfe49a5
 TERMUX_PKG_AUTO_UPDATE=true
 TERMUX_PKG_DEPENDS="libc++, libandroid-spawn"
 
--------------------

bump(main/ninja): 1.12.0


Not committing to Git!

INFO: Trying to build package.
INFO: Package ninja exists in main repo.
Running container 'termux-package-builder' from image 'ghcr.io/termux/package-builder'...
termux - building ninja for arch aarch64...
Downloading https://packages-cf.termux.dev/apt/termux-main/dists/stable/Release
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 13134  100 13134    0     0  50532      0 --:--:-- --:--:-- --:--:-- 50710
Downloading https://packages-cf.termux.dev/apt/termux-main/dists/stable/Release.gpg
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   833  100   833    0     0   1959      0 --:--:-- --:--:-- --:--:--  1960
gpg: Signature made Thu 11 Apr 2024 12:03:49 PM UTC
gpg:                using RSA key CC72CF8BA7DBFA0182877D045A897D96E57CF20C
gpg: Good signature from "Termux Releases (Termux automatic builds) <contact@termux.dev>" [ultimate]
gpg:                 aka "Termux Releases (Termux automatic builds) <contact@termux.org>" [ultimate]
Downloading https://packages-cf.termux.dev/apt/termux-root/dists/root/Release
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 13319  100 13319    0     0  16926      0 --:--:-- --:--:-- --:--:-- 16923
Downloading https://packages-cf.termux.dev/apt/termux-root/dists/root/Release.gpg
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   833  100   833    0     0   3684      0 --:--:-- --:--:-- --:--:--  3702
gpg: Signature made Sun 07 Apr 2024 06:56:25 PM UTC
gpg:                using RSA key CC72CF8BA7DBFA0182877D045A897D96E57CF20C
gpg: Good signature from "Termux Releases (Termux automatic builds) <contact@termux.dev>" [ultimate]
gpg:                 aka "Termux Releases (Termux automatic builds) <contact@termux.org>" [ultimate]
Downloading https://packages-cf.termux.dev/apt/termux-x11/dists/x11/Release
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 13120  100 13120    0     0  76965      0 --:--:-- --:--:-- --:--:-- 77176
Downloading https://packages-cf.termux.dev/apt/termux-x11/dists/x11/Release.gpg
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   833  100   833    0     0   1874      0 --:--:-- --:--:-- --:--:--  1876
gpg: Signature made Thu 11 Apr 2024 12:11:23 PM UTC
gpg:                using RSA key CC72CF8BA7DBFA0182877D045A897D96E57CF20C
gpg: Good signature from "Termux Releases (Termux automatic builds) <contact@termux.dev>" [ultimate]
gpg:                 aka "Termux Releases (Termux automatic builds) <contact@termux.org>" [ultimate]
Downloading dependency libc++@26b if necessary...
Found libc++ in https://packages-cf.termux.dev/apt/termux-main/dists/stable
extracting libc++ to /home/builder/.termux-build/_cache-aarch64...
Downloading dependency libandroid-spawn@0.3 if necessary...
Found libandroid-spawn in https://packages-cf.termux.dev/apt/termux-main/dists/stable
Downloading https://packages-cf.termux.dev/apt/termux-main/pool/main/liba/libandroid-spawn/libandroid-spawn_0.3_aarch64.deb
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 15216  100 15216    0     0   106k      0 --:--:-- --:--:-- --:--:--  106k
extracting libandroid-spawn to /home/builder/.termux-build/_cache-aarch64...
Downloading https://github.com/ninja-build/ninja/archive/v1.12.0.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0

100  234k    0  234k    0     0   700k      0 --:--:-- --:--:-- --:--:--  700k
Applying patch: src-subprocess-posix.cc.patch
wrote build.ninja.
Downloading https://github.com/ninja-build/ninja/releases/download/v1.10.2/ninja-linux.zip
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0

100  100k  100  100k    0     0   329k      0 --:--:-- --:--:-- --:--:--  329k
Archive:  /home/builder/.termux-build/ninja/tmp/ninja-1.10.2.zip
  inflating: /home/builder/.termux-build/_cache/ninja-1.10.2/ninja  
[1/34] RE2C /home/builder/.termux-build/ninja/src/src/depfile_parser.cc
[2/34] RE2C /home/builder/.termux-build/ninja/src/src/lexer.cc
[3/34] INLINE build/browse_py.h
[4/34] CXX build/browse.o
[5/34] CXX build/depfile_parser.o
[6/34] CXX build/lexer.o
[7/34] CXX build/clparser.o
[8/34] CXX build/debug_flags.o
[9/34] CXX build/clean.o
[10/34] CXX build/build_log.o
[11/34] CXX build/disk_interface.o
[12/34] CXX build/build.o
[13/34] CXX build/deps_log.o
[14/34] CXX build/dyndep.o
[15/34] CXX build/edit_distance.o
[16/34] CXX build/dyndep_parser.o
[17/34] CXX build/eval_env.o
[18/34] CXX build/graphviz.o
[19/34] CXX build/json.o
[20/34] CXX build/line_printer.o
[21/34] CXX build/graph.o
[22/34] CXX build/metrics.o
/home/builder/.termux-build/ninja/src/src/metrics.cc:40:19: warning: unused function 'GetFrequency' [-Wunused-function]
constexpr int64_t GetFrequency() {
                  ^
1 warning generated.
[23/34] CXX build/parser.o
[24/34] CXX build/manifest_parser.o
[25/34] CXX build/string_piece_util.o
[26/34] CXX build/missing_deps.o
[27/34] CXX build/state.o
[28/34] CXX build/status.o
[29/34] CXX build/util.o
FAILED: build/util.o 
aarch64-linux-android-clang++ -MMD -MT build/util.o -MF build/util.o.d -g -Wall -Wextra -Wno-deprecated -Wno-missing-field-initializers -Wno-unused-parameter -fno-rtti -fno-exceptions -std=c++11 -fvisibility=hidden -pipe '-DNINJA_PYTHON="python3"' -O2 -DNDEBUG -fdiagnostics-color -DUSE_PPOLL -DNINJA_HAVE_BROWSE -I.  -fstack-protector-strong -Oz  -fstack-protector-strong -Oz  -I/data/data/com.termux/files/usr/include -c /home/builder/.termux-build/ninja/src/src/util.cc -o build/util.o
/home/builder/.termux-build/ninja/src/src/util.cc:899:18: error: variable has incomplete type 'struct sysinfo'
  struct sysinfo si;
                 ^
/home/builder/.termux-build/ninja/src/src/util.cc:899:10: note: forward declaration of 'sysinfo'
  struct sysinfo si;
         ^
/home/builder/.termux-build/ninja/src/src/util.cc:900:20: error: invalid operands to binary expression ('sysinfo' and 'int')
  if (sysinfo(&si) != 0)
      ~~~~~~~~~~~~ ^  ~
/home/builder/.termux-build/_cache/android-r26b-api-24-v3/bin/../sysroot/usr/include/c++/v1/__threading_support:664:6: note: candidate function not viable: cannot convert argument of incomplete type 'sysinfo' to '__thread_id' for 1st argument
bool operator!=(__thread_id __x, __thread_id __y) _NOEXCEPT {
     ^
/home/builder/.termux-build/_cache/android-r26b-api-24-v3/bin/../sysroot/usr/include/c++/v1/__utility/pair.h:457:1: note: candidate template ignored: could not match 'pair<_T1, _T2>' against 'sysinfo'
operator!=(const pair<_T1,_T2>& __x, const pair<_T1,_T2>& __y)
^
/home/builder/.termux-build/_cache/android-r26b-api-24-v3/bin/../sysroot/usr/include/c++/v1/__ios/fpos.h:73:6: note: candidate template ignored: could not match 'fpos<_StateT>' against 'sysinfo'
bool operator!=(const fpos<_StateT>& __x, const fpos<_StateT>& __y) {
     ^
/home/builder/.termux-build/_cache/android-r26b-api-24-v3/bin/../sysroot/usr/include/c++/v1/__iterator/reverse_iterator.h:235:1: note: candidate template ignored: could not match 'reverse_iterator<_Iter1>' against 'sysinfo'
operator!=(const reverse_iterator<_Iter1>& __x, const reverse_iterator<_Iter2>& __y)
^
/home/builder/.termux-build/_cache/android-r26b-api-24-v3/bin/../sysroot/usr/include/c++/v1/__iterator/wrap_iter.h:189:6: note: candidate template ignored: could not match '__wrap_iter<_Iter1>' against 'sysinfo'
bool operator!=(const __wrap_iter<_Iter1>& __x, const __wrap_iter<_Iter1>& __y) _NOEXCEPT
     ^
/home/builder/.termux-build/_cache/android-r26b-api-24-v3/bin/../sysroot/usr/include/c++/v1/__iterator/wrap_iter.h:196:6: note: candidate template ignored: could not match '__wrap_iter<_Iter1>' against 'sysinfo'
bool operator!=(const __wrap_iter<_Iter1>& __x, const __wrap_iter<_Iter2>& __y) _NOEXCEPT
     ^
/home/builder/.termux-build/_cache/android-r26b-api-24-v3/bin/../sysroot/usr/include/c++/v1/__memory/allocator.h:269:6: note: candidate template ignored: could not match 'allocator<_Tp>' against 'sysinfo'
bool operator!=(const allocator<_Tp>&, const allocator<_Up>&) _NOEXCEPT {return false;}
     ^
/home/builder/.termux-build/_cache/android-r26b-api-24-v3/bin/../sysroot/usr/include/c++/v1/tuple:1582:1: note: candidate template ignored: could not match 'tuple<_Tp...>' against 'sysinfo'
operator!=(const tuple<_Tp...>& __x, const tuple<_Up...>& __y)
^
/home/builder/.termux-build/_cache/android-r26b-api-24-v3/bin/../sysroot/usr/include/c++/v1/string_view:815:6: note: candidate template ignored: could not match 'basic_string_view<_CharT, _Traits>' against 'sysinfo'
bool operator!=(basic_string_view<_CharT, _Traits> __lhs, basic_string_view<_CharT, _Traits> __rhs) _NOEXCEPT
     ^
/home/builder/.termux-build/_cache/android-r26b-api-24-v3/bin/../sysroot/usr/include/c++/v1/string_view:824:6: note: candidate template ignored: could not match 'basic_string_view<_CharT, _Traits>' against 'sysinfo'
bool operator!=(basic_string_view<_CharT, _Traits> __lhs,
     ^
/home/builder/.termux-build/_cache/android-r26b-api-24-v3/bin/../sysroot/usr/include/c++/v1/string_view:834:6: note: candidate template ignored: could not match 'basic_string_view<_CharT, _Traits>' against 'int'
bool operator!=(__type_identity_t<basic_string_view<_CharT, _Traits> > __lhs,
     ^
/home/builder/.termux-build/_cache/android-r26b-api-24-v3/bin/../sysroot/usr/include/c++/v1/__memory/unique_ptr.h:529:1: note: candidate template ignored: could not match 'unique_ptr<_T1, _D1>' against 'sysinfo'
operator!=(const unique_ptr<_T1, _D1>& __x, const unique_ptr<_T2, _D2>& __y) {return !(__x == __y);}
^
/home/builder/.termux-build/_cache/android-r26b-api-24-v3/bin/../sysroot/usr/include/c++/v1/__memory/unique_ptr.h:589:1: note: candidate template ignored: could not match 'unique_ptr<_T1, _D1>' against 'sysinfo'
operator!=(const unique_ptr<_T1, _D1>& __x, nullptr_t) _NOEXCEPT
^
/home/builder/.termux-build/_cache/android-r26b-api-24-v3/bin/../sysroot/usr/include/c++/v1/__memory/unique_ptr.h:597:1: note: candidate template ignored: could not match 'unique_ptr<_T1, _D1>' against 'int'
operator!=(nullptr_t, const unique_ptr<_T1, _D1>& __x) _NOEXCEPT
^
/home/builder/.termux-build/_cache/android-r26b-api-24-v3/bin/../sysroot/usr/include/c++/v1/__iterator/istream_iterator.h:96:1: note: candidate template ignored: could not match 'istream_iterator<_Tp, _CharT, _Traits, _Distance>' against 'sysinfo'
operator!=(const istream_iterator<_Tp, _CharT, _Traits, _Distance>& __x,
^
/home/builder/.termux-build/_cache/android-r26b-api-24-v3/bin/../sysroot/usr/include/c++/v1/__iterator/istreambuf_iterator.h:112:6: note: candidate template ignored: could not match 'istreambuf_iterator<_CharT, _Traits>' against 'sysinfo'
bool operator!=(const istreambuf_iterator<_CharT,_Traits>& __a,
     ^
/home/builder/.termux-build/_cache/android-r26b-api-24-v3/bin/../sysroot/usr/include/c++/v1/__iterator/move_iterator.h:245:6: note: candidate template ignored: could not match 'move_iterator<_Iter1>' against 'sysinfo'
bool operator!=(const move_iterator<_Iter1>& __x, const move_iterator<_Iter2>& __y)
     ^
/home/builder/.termux-build/_cache/android-r26b-api-24-v3/bin/../sysroot/usr/include/c++/v1/__memory/shared_ptr.h:1260:1: note: candidate template ignored: could not match 'shared_ptr<_Tp>' against 'sysinfo'
operator!=(const shared_ptr<_Tp>& __x, const shared_ptr<_Up>& __y) _NOEXCEPT
^
/home/builder/.termux-build/_cache/android-r26b-api-24-v3/bin/../sysroot/usr/include/c++/v1/__memory/shared_ptr.h:1335:1: note: candidate template ignored: could not match 'shared_ptr<_Tp>' against 'sysinfo'
operator!=(const shared_ptr<_Tp>& __x, nullptr_t) _NOEXCEPT
^
/home/builder/.termux-build/_cache/android-r26b-api-24-v3/bin/../sysroot/usr/include/c++/v1/__memory/shared_ptr.h:1343:1: note: candidate template ignored: could not match 'shared_ptr<_Tp>' against 'int'
operator!=(nullptr_t, const shared_ptr<_Tp>& __x) _NOEXCEPT
^
/home/builder/.termux-build/_cache/android-r26b-api-24-v3/bin/../sysroot/usr/include/c++/v1/array:398:1: note: candidate template ignored: could not match 'array<_Tp, _Size>' against 'sysinfo'
operator!=(const array<_Tp, _Size>& __x, const array<_Tp, _Size>& __y)
^
/home/builder/.termux-build/_cache/android-r26b-api-24-v3/bin/../sysroot/usr/include/c++/v1/unordered_map:1910:1: note: candidate template ignored: could not match 'unordered_map<_Key, _Tp, _Hash, _Pred, _Alloc>' against 'sysinfo'
operator!=(const unordered_map<_Key, _Tp, _Hash, _Pred, _Alloc>& __x,
^
/home/builder/.termux-build/_cache/android-r26b-api-24-v3/bin/../sysroot/usr/include/c++/v1/unordered_map:2624:1: note: candidate template ignored: could not match 'unordered_multimap<_Key, _Tp, _Hash, _Pred, _Alloc>' against 'sysinfo'
operator!=(const unordered_multimap<_Key, _Tp, _Hash, _Pred, _Alloc>& __x,
^
/home/builder/.termux-build/_cache/android-r26b-api-24-v3/bin/../sysroot/usr/include/c++/v1/vector:3245:1: note: candidate template ignored: could not match 'vector<_Tp, _Allocator>' against 'sysinfo'
operator!=(const vector<_Tp, _Allocator>& __x, const vector<_Tp, _Allocator>& __y)
^
/home/builder/.termux-build/_cache/android-r26b-api-24-v3/bin/../sysroot/usr/include/c++/v1/__functional/function.h:1199:1: note: candidate template ignored: could not match 'function<_Rp (_ArgTypes...)>' against 'sysinfo'
operator!=(const function<_Rp(_ArgTypes...)>& __f, nullptr_t) _NOEXCEPT {return (bool)__f;}
^
/home/builder/.termux-build/_cache/android-r26b-api-24-v3/bin/../sysroot/usr/include/c++/v1/__functional/function.h:1204:1: note: candidate template ignored: could not match 'function<_Rp (_ArgTypes...)>' against 'int'
operator!=(nullptr_t, const function<_Rp(_ArgTypes...)>& __f) _NOEXCEPT {return (bool)__f;}
^
/home/builder/.termux-build/_cache/android-r26b-api-24-v3/bin/../sysroot/usr/include/c++/v1/string:4094:1: note: candidate template ignored: could not match 'basic_string<_CharT, _Traits, _Allocator>' against 'sysinfo'
operator!=(const basic_string<_CharT,_Traits,_Allocator>& __lhs,
^
/home/builder/.termux-build/_cache/android-r26b-api-24-v3/bin/../sysroot/usr/include/c++/v1/string:4103:1: note: candidate template ignored: could not match 'const _CharT *' against 'sysinfo'
operator!=(const _CharT* __lhs,
^
/home/builder/.termux-build/_cache/android-r26b-api-24-v3/bin/../sysroot/usr/include/c++/v1/string:4112:1: note: candidate template ignored: could not match 'basic_string<_CharT, _Traits, _Allocator>' against 'sysinfo'
operator!=(const basic_string<_CharT, _Traits, _Allocator>& __lhs,
^
/home/builder/.termux-build/ninja/src/src/util.cc:902:22: error: use of undeclared identifier 'SI_LOAD_SHIFT'
  return 1.0 / (1 << SI_LOAD_SHIFT) * si.loads[0];
                     ^
3 errors generated.
[30/34] CXX build/version.o
[31/34] CXX build/subprocess-posix.o
[32/34] CXX build/ninja.o
ninja: build stopped: subcommand failed.
ERROR: failed to build.</pre>
</details>

<hr>
<i>
Above error occured when I last tried to update at 2024-04-11 18:22:46 UTC.<br>
Run ID: <a href="https://github.com/termux/termux-packages/actions/runs/8651233240">8651233240</a><br><br>
<b>Note:</b> Automatic updates will be disabled until this issue is resolved.<br>
</i>

--- a/packages/ninja/build.sh
+++ b/packages/ninja/build.sh
@@ -2,11 +2,11 @@ TERMUX_PKG_HOMEPAGE=https://ninja-build.org
 TERMUX_PKG_DESCRIPTION="A small build system with a focus on speed"
 TERMUX_PKG_LICENSE="Apache-2.0"
 TERMUX_PKG_MAINTAINER="@termux"
-TERMUX_PKG_VERSION="1.11.1"
-TERMUX_PKG_SRCURL=https://github.com/ninja-build/ninja/archive/v${TERMUX_PKG_VERSION}.tar.gz
-TERMUX_PKG_SHA256=31747ae633213f1eda3842686f83c2aa1412e0f5691d1c14dbbcc67fe7400cea
+TERMUX_PKG_VERSION="1.12.0"
+TERMUX_PKG_SRCURL=https://github.com/ninja-build/ninja/archive/refs/tags/v${TERMUX_PKG_VERSION}.tar.gz
+TERMUX_PKG_SHA256=8b2c86cd483dc7fcb7975c5ec7329135d210099a89bc7db0590a07b0bbfe49a5
+TERMUX_PKG_DEPENDS="libandroid-spawn, libc++"
 TERMUX_PKG_AUTO_UPDATE=true
-TERMUX_PKG_DEPENDS="libc++, libandroid-spawn"
 
 termux_step_pre_configure() {
 	CXXFLAGS+=" $CPPFLAGS"

--- a/packages/ninja/src-util.cc.patch
+++ b/packages/ninja/src-util.cc.patch
@@ -1,0 +1,29 @@
+--- a/src/util.cc
++++ b/src/util.cc
+@@ -47,7 +47,7 @@
+ #include <sys/loadavg.h>
+ #elif defined(_AIX) && !defined(__PASE__)
+ #include <libperfstat.h>
+-#elif defined(linux) || defined(__GLIBC__)
++#elif defined(__linux__) || defined(__GLIBC__)
+ #include <sys/sysinfo.h>
+ #include <fstream>
+ #include <map>
+@@ -589,7 +589,7 @@
+   return stripped;
+ }
+ 
+-#if defined(linux) || defined(__GLIBC__)
++#if defined(__linux__) || defined(__GLIBC__)
+ std::pair<int64_t, bool> readCount(const std::string& path) {
+   std::ifstream file(path.c_str());
+   if (!file.is_open())
+@@ -789,7 +789,7 @@
+ #else
+   int cgroupCount = -1;
+   int schedCount = -1;
+-#if defined(linux) || defined(__GLIBC__)
++#if defined(__linux__) || defined(__GLIBC__)
+   cgroupCount = ParseCPUFromCGroup();
+ #endif
+   // The number of exposed processors might not represent the actual number of

